pkg/controller/model/grafanaDeployment.go

```diff
@@ -366,6 +366,10 @@ func getDeploymentSpec(cr *v1alpha1.Grafana, configHash, plugins, dsHash string)
 				Annotations: getPodAnnotations(cr),
 			},
 			Spec: v13.PodSpec{
+				NodeSelector:       cr.Spec.Deployment.NodeSelector,
```

Comment:
This will fail when `deployment` is not defined in the CR (because deployment itself is a [pointer](https://github.com/integr8ly/grafana-operator/blob/master/pkg/apis/integreatly/v1alpha1/grafana_types.go#L27) and can be null). I'd suggest using a function that checks if `deployment` is defined and otherwise just returns nil for those properties. See for example replicas: https://github.com/integr8ly/grafana-operator/blob/234b8bd58f335d0275a651e612f7c283e885ebf8/pkg/controller/model/grafanaDeployment.go#L40

Reply:
@pb82 got it. Ill add this, makes sense and i guess thats a best practice. I am travelling so please expect some delay.

Reply:
@pb82 resolved